src/main/java/io/zksync/signer/Create2EthSigner.java

```diff
@@ -0,0 +1,154 @@
+package io.zksync.signer;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+
+import org.bouncycastle.util.Arrays;
+import org.web3j.crypto.Hash;
+import org.web3j.protocol.Web3j;
+import org.web3j.tx.ReadonlyTransactionManager;
+import org.web3j.tx.TransactionManager;
+import org.web3j.utils.Numeric;
+
+import io.zksync.domain.auth.ChangePubKeyCREATE2;
+import io.zksync.domain.swap.Order;
+import io.zksync.domain.token.Token;
+import io.zksync.domain.token.TokenId;
+import io.zksync.domain.transaction.ChangePubKey;
+import io.zksync.domain.transaction.ZkSyncTransaction;
+import io.zksync.exception.ZkSyncException;
+
+public class Create2EthSigner implements EthSigner<ChangePubKeyCREATE2> {
+    private final TransactionManager transactionManager;
+
+    private ChangePubKeyCREATE2 authData;
+    private String address;
+
+    private Create2EthSigner(String zkSyncAddress, String address, ChangePubKeyCREATE2 create2Data, TransactionManager transactionManager) {
```

Comment:
Do we need `zkSyncAddress` parameter here?

Reply:
Good call! 

@MaximFischuk is `zkSyncAddress` ever different from `address`?

Reply:
This class can be constructed only used factory methods. Also private constructor cannot be called from anywhere except members of this class. Thus this constructor used only as internal fields initializer

Reply:
I understand, but is it possible that the `zkSyncAddress` is different from `address`? I mean why do we need to initialize these two fields in the first place? It is not critical, but it is weird

Reply:
> I understand, but is it possible that the `zkSyncAddress` is different from `address`?
In this case it's impossible (except some hacks like construct using reflection :-) )

> I mean why do we need to initialize these two fields in the first place?
I guess `zkSyncAddress` parameter of this constructor is useless, I'll remove it. I'll left only ethereum address cause it used for `getAddress` method